src/nft/components/view/NFTImage.tsx

```diff
@@ -0,0 +1,91 @@
+import { useCallback, useEffect, useState } from 'react';
+import { defaultNFTSvg } from '../../../internal/svg/defaultNFTSvg';
+import { cn } from '../../../styles/theme';
+import type { NFTError } from '../../types';
+import { useNFTContext } from '../NFTProvider';
+
+type NFTImageReact = {
+  className?: string;
+  onLoading?: (mediaUrl: string) => void;
+  onLoaded?: () => void;
+  onError?: (error: NFTError) => void;
+};
+
+export function NFTImage({
+  className,
+  onLoading,
+  onLoaded,
+  onError,
```

Comment:
should these just be `onStatus`?

Reply:
I think I'd need to basically create a pseudo lifecycle status to change to onStatus.  I could handle lifecycle status in the children, but I was trying to keep the mediaLoaded/success logic centralized in one place.